src/Sulu/Bundle/DocumentManagerBundle/Bridge/DocumentInspector.php

```diff
@@ -240,8 +253,17 @@ public function getLocalizedUrlsForPage(BasePageDocument $page)
         $rlpProperty = $structure->getPropertyByTagName('sulu.rlp');
 
         foreach ($webspace->getAllLocalizations() as $localization) {
-            $locale = $localization->getLocalization();
-            $stageName = $this->encoder->localizedSystemName(WorkflowStageSubscriber::WORKFLOW_STAGE_FIELD, $locale);
+            $resolvedLocale = $localization->getLocalization();
```

Comment:
This method would be much simpler if it worked with the documents, but this is faster. In anycase, we should remove all of the URL stuff from the dcouments and move that logic to the `Router`